IdentityCore/src/cache/token/MSIDCredentialCacheItem.m

```diff
@@ -194,29 +197,7 @@ - (NSDictionary *)jsonDictionary
 
 - (MSIDBaseToken *)tokenWithType:(MSIDCredentialType)credentialType
 {
-    switch (credentialType)
-    {
-        case MSIDAccessTokenType:
-        {
-            return [[MSIDAccessToken alloc] initWithTokenCacheItem:self];
-        }
-        case MSIDRefreshTokenType:
-        {
-            return [[MSIDRefreshToken alloc] initWithTokenCacheItem:self];
-        }
-        case MSIDLegacySingleResourceTokenType:
-        {
-            return [[MSIDLegacySingleResourceToken alloc] initWithTokenCacheItem:self];
-        }
-        case MSIDIDTokenType:
-        {
-            return [[MSIDIdToken alloc] initWithTokenCacheItem:self];
-        }
-        default:
-            return [[MSIDBaseToken alloc] initWithTokenCacheItem:self];
-    }
-    
-    return nil;
+    return [MSIDCredentialCacheItemUtil tokenWithType:credentialType credential:self];
```

Comment:
This will still cause a problem for CPP if you don't want to include MSIDCredentialCacheItemUtil, right?

Reply:
On our side I have a do-nothing stub.

Reply:
I'm even fine with you putting an #ifdef around that call so you don't need to add a stub.

Reply:
I suggest moving this method to a category, like MSIDCredentialCacheItem+ MSIDBaseToken. We have only a few places where we need it in common core, and one place in ADAL. This way you would not need to add stub on your side.

Reply:
I'm fine with a category too.

Reply:
changed to a simple #ifdef

Reply:
Based on further feedback, I replace this with a category, `MSIDCredentialCacheItem+ MSIDBaseToken`.